src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -339,6 +339,10 @@ private function set($type, $id)
     private function createAutowiredDefinition(\ReflectionClass $typeHint, $id)
     {
         if (isset($this->ambiguousServiceTypes[$typeHint->name])) {
+            if ($this->container->has($typeHint->name)) {
```

Comment:
Can't this be moved [here](https://github.com/theofidry/symfony/blob/24d3ff4fcd9916f38a936ef62a2192c534256630/src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php#L196-L200) to avoid building the map when not necessary?